interfaces/builtin/utils.go

```diff
@@ -96,3 +97,26 @@ func sanitizeSlotReservedForOSOrApp(iface interfaces.Interface, slot *snap.SlotI
 	}
 	return nil
 }
+
+// determine if permanent slot side is provided by the system
+// on classic system some implicit slots can be provided by system or by
+// application snap e.g. avahi (it can be installed as deb or snap)
+// - slot owned by the system (core,snapd snap)  usually requires no action
+// - slot owned by application snap typically requires rules update
+func isSlotSystemSlot(slot *snap.SlotInfo) bool {
+	if release.OnClassic &&
+		(slot.Snap.Type == snap.TypeOS || slot.Snap.Type == snap.TypeSnapd) {
```

Comment:
This is good but probably needs to include a check for the snap name "snapd" - we don't use the snapd type yet (mostly because in the past the store was not ready and now I think it is but we dropped the ball).

Reply:
@mvo5 instead of adding name check to many different places, wouldn't it be cleaner to simply choose correct enum when populating snap info? So we have name check only in one place?